### PR TITLE
Return the actual bound address, not a hardcoded "localhost".

### PR DESCRIPTION
**Pull request by bemasc**
This address will always be either '127.0.0.1' or '0.0.0.0',
because nsIUDPSocket::init only supports binding to localhost
or all-interfaces.

Addresses https://github.com/freedomjs/freedom-for-firefox/issues/62 and https://github.com/freedomjs/freedom-for-firefox/issues/67

@trevj

### CODE REVIEW

File: providers/core.udpsocket.js [hunk 1]
```diff
@@ -10,7 +10,10 @@ UDP_Firefox.prototype.bind = function(address, port, continuation) {
     port = -1;
   }
   try {
-    this._nsIUDPSocket.init(port, false);
+    var isLocal = address === '127.0.0.1' ||
+        address === 'localhost' ||
+        address.match(/^(0*:)+0*1$/);
```
Comment (soycode) [inline providers/core.udpsocket.js]: Why the 1 at the end of this regex?

Comment (bemasc) [inline providers/core.udpsocket.js]: OK, I added a comment.  Does it help?

**Comment by soycode** [comment]
:+1:

**Comment by bemasc** [comment]
I had a conversation with Trevor and changed the logic here to be much more restrictive.  Instead of binding to a different address than the one the caller requested, it now returns an error code.

However, this will actually break uProxy ... so please don't press "Merge" so we have time to update uProxy first!

**Comment by bemasc** [comment]
uProxy change here: https://github.com/uProxy/uproxy-lib/pull/212

File: providers/core.udpsocket.js [hunk 2]
```diff
@@ -23,7 +38,7 @@ UDP_Firefox.prototype.bind = function(address, port, continuation) {
 
 UDP_Firefox.prototype.getInfo = function(continuation) {
   var returnValue = {
-    localAddress: "localhost",
+    localAddress: this._nsIUDPSocket.localAddr.address,
     localPort: this._nsIUDPSocket.port
```
Comment (soycode) [inline providers/core.udpsocket.js]: It may be good to pull port from `this._nsIUDPSocket.localAddr.port` for consistency - from logging the tests it looks to have the same value as `this._nsIUDPSocket.port`, so it should work the same, but look more consistent and maybe inoculate us from weird edge cases.

**Comment by soycode** [comment]
Consider the port suggestion, but aside from that :+1:

File: providers/core.udpsocket.js (hunk 2 above)
Comment (bemasc) [inline providers/core.udpsocket.js]: Done.
